src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/includeClassifier.js

```diff
@@ -0,0 +1,56 @@
+import Url from 'url-parse';
+
+const supportedTypes = {
+  indepthtoolkit: 'idt1',
+  idt2: 'idt2',
+  include: 'vj',
+  'news/special': 'vj',
+  'market-data': 'vj',
+  'smallprox/include': 'vj',
+};
+
+const ampSupported = href => {
+  // An amp-image query parameter on the include path indicates an AMP version of the include is available
+  const hasAmpImageQueryString = new Url(href, true).query['amp-image'];
+  return !!hasAmpImageQueryString;
+};
+
+const includeClassifier = ({ href, pathname }) => {
+  // This determines if the href has a leading '/'
+  const hrefTypePostion = () => (href.indexOf('/') === 0 ? 1 : 0);
+
+  // This checks if the supportedType is in the correct position of the href
+  const hrefIsSupported = () => supportedType =>
+    href.startsWith(supportedType, hrefTypePostion());
+
+  // This extracts the type from the href
+  const typeExtraction = Object.keys(supportedTypes).find(
+    hrefIsSupported(href),
+  );
+
+  // This determines if the type is supported and returns the include type name
+  const includeType = supportedTypes[typeExtraction];
+
+  const pathnameIsAmp = pathname.endsWith('.amp');
+
+  if (includeType === 'vj') {
+    if (!pathnameIsAmp) {
+      return { includeType, classification: 'vj-include-canonical' };
+    }
+
+    if (ampSupported(href)) {
+      return { includeType, classification: 'vj-include-supports-amp' };
+    }
+
+    return { includeType, classification: 'vj-include-not-supporting-amp' };
+  }
+  if (includeType) {
+    const platform = pathnameIsAmp ? 'amp' : 'canonical';
+    const classification = `${includeType}-${platform}`;
+
+    return { includeType, classification };
+  }
+  return { includeType: null, classification: 'not-supported' };
```

Comment:
Not sure if I missed something  how is the include block being returned differently for `not-supported` and `vj-include-not-supporting-amp`, I reckon if this could be simplified as such:
 
```
  const platform = pathnameIsAmp ? 'amp' : 'canonical';
  const classification = `${includeType}-${platform}`;

  if (!pathnameIsAmp || !ampSupported(href)) {
    return { includeType, classification };
  }

  if (ampSupported(href)) {
    return { includeType, classification: 'vj-include-supports-amp' };
  }

  return { includeType: includeType || null, classification: 'not-supported' };
```

we could group `not-supported` and `vj-include-not-supporting-amp` both as `not-supported`

Reply:
For the `vj-include-not-supporting-amp` classification the `includeType` will be `vj` - so it should work on canonical just not on amp. Whereas the `not-supported` classification is for includes which are not included in `supportedTypes` and so won't work on amp or canonical, so I think it is worth keeping the distinction.

Reply:
That sounds a better approach @rhenshaw56, we don't treat `vj-include-not-supporting-amp` any different to `not-supported` at the converter level. Also will simplify the 'not supported logic' we've discussed here: https://github.com/bbc/simorgh/pull/6873#discussion_r440227803